rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/preprocess.kt

```diff
@@ -0,0 +1,40 @@
+package com.smeup.rpgparser.parsing.facade
+
+import com.smeup.rpgparser.execution.MainExecutionContext
+import com.smeup.rpgparser.interpreter.PreprocessingLogEnd
+import com.smeup.rpgparser.interpreter.PreprocessingLogStart
+import java.io.BufferedReader
+import java.io.InputStream
+import kotlin.system.measureTimeMillis
+
+internal fun InputStream.preprocess(
+    findCopy: (copyId: CopyId) -> String?,
+    onStartInclusion: (copyId: CopyId, start: Int) -> Unit = { _: CopyId, _: Int -> },
+    onEndInclusion: (end: Int) -> Unit = { _: Int -> },
+    beforeInclusion: (copyId: CopyId) -> Boolean = { true },
+    resolveDirectives: Boolean = true
+): String {
+    val programName = getExecutionProgramNameWithNoExtension()
+    MainExecutionContext.log(PreprocessingLogStart(programName = programName))
+    var preprocessed: String
+    measureTimeMillis {
+        preprocessed = bufferedReader().use(BufferedReader::readText).includesCopy(
+            findCopy = findCopy,
+            onStartInclusion = onStartInclusion,
+            onEndInclusion = onEndInclusion,
+            beforeInclusion = beforeInclusion
+        )
+        if (resolveDirectives) {
```

Comment:
Simply, I don't understand well what could be the issue. In my simple mind, the interpreter executes resolved code line by line, so I don't understand why there is a necessity to register the relative line of statements in the copies.
So, I trust you.

Reply:
It is necessary for the following reasons:
- in case of error (syntax or runtime) when the program is not a toy or silly program, the interpreted program is the result of inner copy inclusions and contains thousand of statements and thousands of lines. If the error is in absolute line 5965 (for instance) it is very difficult to understand which is the copy involved, becasue undestanding the copy involved is crucial to make a decent problem determination
- another crucial aspect regards the deugging, the user with jardis can put a break-point inside a copy and following step by step the program execution just like the program is written, not like the program is known by jariko (a blob of source line of codes containing the exploded copy)

Reply:
Ok, if you believe and if you think I could be able to do it, I could try to manage CopyBlock update after compiler directives resolution.  Let me know.

Reply:
Maybe there could be a simple solution. The compiler directives resolution engine might not remove rows from sources but simply comment them. This way, all the problems related to the source line index could be resolved.
What do you think about this solution?

Reply:
The lines not to be used after executing the compiler directives are now kept in the source but transformed into comments. The resulting source is not very readable, but the solution seems to work. All tests are now executed successfully.